e2e/stress/IoTClientPerf/Scenarios/ServiceClientScenarios.cs

```diff
@@ -101,22 +111,33 @@ protected async Task CallMethodAsync(CancellationToken ct)
 
             try
             {
+                string deviceId = Configuration.Stress.GetDeviceNameById(_id, _authType);
+
                 var methodCall = new CloudToDeviceMethod(TestMethodName);
-                Task t = s_sc.InvokeDeviceMethodAsync(Configuration.Stress.GetDeviceNameById(_id, _authType), methodCall);
+                methodCall.SetPayloadJson(_methodPayload);
+                Task<CloudToDeviceMethodResult> t = s_sc.InvokeDeviceMethodAsync(Configuration.Stress.GetDeviceNameById(_id, _authType), methodCall);
                 _mMethod.ScheduleTime = _swMethod.ElapsedMilliseconds;
 
                 _swMethod.Restart();
-                await t.ConfigureAwait(false);
+                CloudToDeviceMethodResult result = await t.ConfigureAwait(false);
+
+                // Check method result.
+                if (result.Status != MethodPassStatus)
+                {
+                    throw new InvalidOperationException($"Method Failed status={result.Status} Payload:{result.GetPayloadAsJson()}");
+                }
             }
             catch (Exception ex)
             {
                 _mMethod.ErrorMessage = $"{ex.GetType().Name} - {ex.Message}";
                 exInfo = ExceptionDispatchInfo.Capture(ex);
```

Comment:
just throw. No need to capture.

Reply:
That won't work: I need to capture so that I can write the exception to the log.

Reply:
Discussed offline: it's now in the finally block so it should always execute - no need to capture.